### PR TITLE
Issue 53 - Build script to build and move all build assets into one directory

### PR DESCRIPTION
**Pull request by mmBabol**
Created a build script that will build all of our assets and move everything into a `deploy` directory.

Travis will run the script before deployment

Finally, added the deploy directory to gitignore

### CODE REVIEW

**Comment by dsych** [comment]
@mmBabol there is a commit from #69 , you should rebase it once that PR lands.

File: .travis.yml [hunk 1]
```diff
@@ -14,19 +14,15 @@ before_script:
...
+# Do the entire build
+  - run npm build
 
 deploy:
   provider: pages
   skip-cleanup: true
   github-token: $GITHUB_PAGES  # Set in the settings page of your repository, as a secure variable
   keep-history: false
-  local-dir: src/docs/website/build/next
+  local-dir: deploy
```
Comment (humphd) [inline .travis.yml]: Let's do what they do in Rome, and use `dist`.

File: tools/build.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env node
+
+const commandRunner = require('./run-command');
+
+const build = 'npm run build';
+const build_db = 'npm run build -- --public-url=/"$GH-PAGES/"';
+
+commandRunner('mkdir -p deploy && rm -R deploy/', '.')
```
Comment (humphd) [inline tools/build.js]: Let's use the `mkdirp` node module.

Comment (humphd) [reply tools/build.js]: Also, let's use `rimraf` from npm to first `clean` the `dist/` dir, then you can do `mkdirp` to create it again.

**Comment by mmBabol** [comment]
I've completed the changes for the build script. I'm now using [mkdirp](https://www.npmjs.com/package/mkdirp) for creating directories, [rimraf](https://www.npmjs.com/package/rimraf) for removing directories, and I'm now copying the source code instead of moving with [copy](https://www.npmjs.com/package/copy).

I'm not entirely sure where to place db code though, I currently have it in `dist/dist`

File: tools/build.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+const build = 'npm run build';
+const build_db = 'npm run build -- --public-url=/"$GH-PAGES/"';
+
+var mkdirp = require('mkdirp');
+var rimraf = require('rimraf');
+var copy = require('copy');
+
+rimraf('./dist', {}, function() {});
+
+mkdirp('dist', function(err) {
```
Comment (dsych) [inline tools/build.js]: This is an async function. We should either use a [sync](https://www.npmjs.com/package/mkdirp#mkdirpsyncdir-opts) version or promisify the async version and link it with the rest of the chain.

File: tools/build.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    )
+    .then(() =>
+        mkdirp('dist/editor', function(err) {
+            if (err) console.error(err);
+            else console.log('Created dist/editor directory');
+        })
+    )
+    .then(() => commandRunner(build, 'src/editor'))
+    .then(() =>
+        copy('./src/editor/dist/**', './dist/edtor', function(err, file) {
```
Comment (dsych) [inline tools/build.js]: Same thing as with mkdirp. We should promisify this and inject it into the promise chain.

Comment (mmBabol) [reply tools/build.js]: I can promisify both rimraf and mkdirp. To do that, would I do something similar to `run-command.js` where I pass the arguments there and then promisify it?

Comment (dsych) [reply tools/build.js]: Yes

**Comment by twigz20** [comment]
@mmBabol Great Job!

**Pull request merged**